pkg/rancher/rancher.go

```diff
@@ -104,7 +104,14 @@ func New(ctx context.Context, clientConfg clientcmd.ClientConfig, opts *Options)
 	}
 
 	if features.MCM.Enabled() && !features.Fleet.Enabled() {
-		return nil, fmt.Errorf("multi-cluster-management features requires feature fleet=true")
+		logrus.Info("fleet can't be turned off when MCM is enabled. Turning on fleet and turning off continuous-delivery")
+		// fleet needs to be turned on. If fleet is not and MCM is on, turn on fleet and turn off continuous-delivery
```

Comment:
nit: I think the info statement makes the comment not necessary

Reply:
@rmweir We need to be able to handle the upgrade case where user has turned off fleet in 2.5 and upgrade to 2.6. If we are still returning the error the upgrade will be broken for these users and it will require extra steps to recover the setup.

Reply:
See this comment https://github.com/rancher/rancher/issues/32808#issuecomment-845480007